content/source/docs/extend: Add Debugging Providers link to SDKv2 Upgrade Guide (#1797)
hashicorp/terraform-website: content/source/docs/extend/guides/v2-upgrade-guide.html.md

connect to it. Providers using `plugin.Debug` will output instructions on how
to tell Terraform how to connect.

> Full details about this topic can be found on the [Debugging Providers page](/docs/extend/debugging.html).

Reviewer: ```suggestion
-> **Note**: You can find more details on the [Debugging Providers page](/docs/extend/debugging.html).
```

Author: Love it!
